pkgs/os-specific/linux/kernel/cachyos-kernels.nix

```
@@ -0,0 +1,155 @@
+{
+  lib,
+  stdenv,
+  fetchFromGitHub,
+  pkgs,
+  buildLinux,
+  ...
+}@args:
+
+let
+  cachyPatches = fetchFromGitHub {
+    owner = "CachyOS";
+    repo = "kernel-patches";
+    rev = "04fb9178158a7714aeb908c7a5310c4e0d6ea4b6";
+    hash = "sha256-KTvZxYy+UgLNg1DYBMrJrRdxDo3N5ususlrHyRBS7C8=";
+  };
+
+  ltsVariant = {
+    version = "6.6.43";
+    source = {
+      url = "mirror://kernel/linux/kernel/v6.x/linux-6.6.43.tar.xz";
+      hash = "sha256-Ctg7Ghp4ChqtlI1VqlXuY8UMYm8tRpELnSGAAo0QCl4=";
+    };
+    patches = [
+      {
+        name = "cachyos-base-all";
+        patch = "${cachyPatches}/6.6/all/0001-cachyos-base-all.patch";
+      }
+      {
+        name = "bore-cachy";
+        patch = "${cachyPatches}/6.6/sched/0001-bore-cachy.patch";
+      }
+    ];
+
+    config = with lib.kernel; {
+
+    };
+  };
+
+  mainVariant = {
+    version = "6.10.2";
+    source = {
+      url = "mirror://kernel/linux/kernel/v6.x/linux-6.10.2.tar.xz";
+      hash = "sha256-c9hSDdnLpaz8XnII52s12XQLiq44IQqSJOMuxMDSm3A=";
+    };
+    patches = [
+      {
+        name = "cachyos-base-all";
+        patch = "${cachyPatches}/6.10/all/0001-cachyos-base-all.patch";
+      }
+      {
+        name = "sched-ext";
+        patch = "${cachyPatches}/6.10/sched/0001-sched-ext.patch";
+      }
+      {
+        name = "bore-cachy-ext";
+        patch = "${cachyPatches}/6.10/sched/0001-bore-cachy-ext.patch";
+      }
+    ];
+
+    config = with lib.kernel; {
+      # CachyOS config
+      CACHY = yes;
+
+      # CPU Scheduler
+      SCHED_CLASS_EXT = yes;
+      SCHED_BORE = yes;
+      MIN_BASE_SLICE_NS = 1000000;
+
+      # LLVM level
+      LTO_NONE = yes;
+
+      # Tick rate
+      HZ_300 = null;
+      HZ_1000 = yes;
+      HZ = 1000;
+
+      NR_CPUS = 320;
+
+      # Tick type
+      HZ_PERIODIC = null;
+      NO_HZ_IDLE = null;
+      CONTEXT_TRACKING_FORCE = null;
+      NO_HZ_FULL_NODEF = yes;
+      NO_HZ_FULL = yes;
+      NO_HZ = yes;
+      NO_HZ_COMMON = yes;
+      CONTEXT_TRACKING = yes;
+
+      # Preempt type
+      PREEMPT_BUILD = yes;
+      PREEMPT_NONE = null;
+      PREEMPT_VOLUNTARY = null;
+      PREEMPT = yes;
+      PREEMPT_COUNT = yes;
+      PREEMPTION = yes;
+      PREEMPT_DYNAMIC = yes;
+
+      # Enable O3
+      CC_OPTIMIZE_FOR_PERFORMANCE = null;
+      CC_OPTIMIZE_FOR_PERFORMANCE_O3 = yes;
+
+      # Enable bbr3
+      TCP_CONG_CUBIC = module;
+      DEFAULT_CUBIC = null;
+      TCP_CONG_BBR = yes;
+      DEFAULT_BBR = yes;
+      EFAULT_TCP_CONG = "bbr";
+
+      # Select THP
+      TRANSPARENT_HUGEPAGE_MADVISE = null;
+      TRANSPARENT_HUGEPAGE_ALWAYS = yes;
+
+      USER_NS = yes;
+    };
```

The cachy kernels include some extra modules, ntsync might be especially useful to some users. They aren't enabled by default.

```suggestion

      I2C_NCT6775 = module;
      V4L2_LOOPBACK = module;
      NTSYNC = yes;
    };
```

---

Can these be enabled via overrides?

---

If we kept them fully disabled and let users override the config, that would trigger a full kernel rebuild instead of just downloading from the nixpkgs cache. It would be best to build them as modules by default - they won't load anyway unless they're added to `boot.kernelModules`. 

I'm not sure about `ntsync`. CachyOS builds it into the kernel, but maybe it would be better if we built it as a module. I suggested `yes` here because that's what cachy does, but a module would probably be safer. I have `NTSYNC = module` in my personal config.